### ISSUE TITLE
Cascading filters in Superset Filter Box

### ISSUE THREAD

**Issue by t0nydean** [apache/incubator-superset#10013]
When you are filtering by more than one element, the values chosen in previous filters should limit the choices in subsequent drop downs
Consider the Country - State - City example....or any hierarchy of values really. 

I would like to have filters applied in sequence, so that if I choose 'USA' from my country filter, I only get options relevant within USA. Then if I choose 'Ohio' from my state drop down, I would only get cities in OH from my 'City' drop down. 

We can create multiple filters that apply to each other but the visual is bad...the filters take up way too much room on the screen. 

I know this has been proposed before ( [5664](https://github.com/apache/incubator-superset/pull/5664) ) but seems like it was shelved dues to some other filtering changes coming, what happened to that?

**Comment by nytai** [comment]
@t0nydean This seems like a feature that would be useful. I can't think of any _good_ reasons we wouldn't want filter to cascade, thought it is a change in behavior. This could probably be wrapped in a config flag if there are use cases for maintaining current behavior. Would you be interested in reviving the old PR (#5664) and working on this? I'd be happy to review and help get this in.

**Comment by t0nydean** [comment]
sorry I am not able to contribute code, the old PR was fully approved if its just a matter of re-requesting it I maybe can do that?

**Comment by nytai** [comment]
That PR is fairly old so it's unlikely it's just work without needing to resolve some merge conflicts. Maybe someone else in the community can take this effort on as we're quite busy with other efforts at the moment

**Comment by JunlinC** [comment]
🏷 filter

**Comment by junlincc** [comment]
#12148
dashboard native filter and cascading filter are available for testing!! thanks in advance for feedback, comments, critics, love and support!!

**Comment by marituhone** [comment]
@junlincc  is the issue get fixed can we cascade filters based on parent filter?

**Comment by rusackas** [comment]
Yes you can... they're called "dependent filters" in the filter config.
<img width="875" alt="Pasted_Image_4_30_24__9_02 AM" src="https://github.com/apache/superset/assets/812905/75f5a070-1db2-4d30-882b-0388f885f083">
